src/components/include/utils/logger.h

```diff
@@ -33,151 +33,110 @@
-  namespace {                                            \
-  CREATE_LOGGERPTR_LOCAL(logger_var, logger_name);       \
+#define SDL_CREATE_LOGGERPTR(component_name)  \
+  namespace {                              \
+  static std::string logger_(component_name); \
   }
 
-#define CREATE_LOGGERPTR_LOCAL(logger_var, logger_name) \
-  log4cxx::LoggerPtr logger_var =                       \
-      log4cxx::LoggerPtr(log4cxx::Logger::getLogger(logger_name));
+#define SDL_CREATE_LOCAL_LOGGERPTR(component_name) \
+  std::string logger_(component_name);
 
-#define INIT_LOGGER(file_name, logs_enabled)           \
-  log4cxx::PropertyConfigurator::configure(file_name); \
-  logger::set_logs_enabled(logs_enabled);
+// special macros to dump logs from queue
+// it's need, for example, when crash happend
+#define SDL_FLUSH_LOGGER() logger::Logger::instance().Flush();
 
 // Logger deinitilization function and macro, need to stop log4cxx writing
 // without this deinitilization log4cxx threads continue using some instances
 // destroyed by exit()
-void deinit_logger();
-#define DEINIT_LOGGER() deinit_logger()
-
-// Logger thread deinitilization macro that need to stop the thread of handling
-// messages for the log4cxx
-#define DELETE_THREAD_LOGGER(logger_var) \
-  logger::delete_log_message_loop_thread(logger_var)
-
-// special macros to dump logs from queue
-// it's need, for example, when crash happend
-#define FLUSH_LOGGER() logger::flush_logger()
-
-#define LOG4CXX_IS_TRACE_ENABLED(logger) logger->isTraceEnabled()
-
-log4cxx_time_t time_now();
-
-#define LOG_WITH_LEVEL(loggerPtr, logLevel, logEvent)                \
-  do {                                                               \
-    if (logger::logs_enabled()) {                                    \
-      if (logger::logger_status != logger::DeletingLoggerThread) {   \
-        if (loggerPtr->isEnabledFor(logLevel)) {                     \
-          std::stringstream accumulator;                             \
-          accumulator << logEvent;                                   \
-          logger::push_log(                                          \
-              loggerPtr,                                             \
-              logLevel,                                              \
-              accumulator.str(),                                     \
-              time_now(),                                            \
-              LOG4CXX_LOCATION,                                      \
-              ::log4cxx::spi::LoggingEvent::getCurrentThreadName()); \
-        }                                                            \
-      }                                                              \
-    }                                                                \
+#define SDL_DEINIT_LOGGER() logger::Logger::instance().DeInit();
+
+#define LOG_WITH_LEVEL(logLevel, logEvent)                             \
+  do {                                                                 \
```

Comment:
@VladSemenyuk don't understand the trick with do-while(false)... Maybe it can be removed at all?